api/api_controller.go

```diff
@@ -1069,6 +1069,14 @@ func (c *Controller) ObjectsUploadObjectHandler() objects.UploadObjectHandler {
 		if err != nil {
 			return objects.NewUploadObjectDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
 		}
+		// check if branch exists - it is still a possibility, but we don't want to upload large object when the branch was not there in the first place
+		branchExists, err := cataloger.IsBranchExists(c.Context(), params.Repository, params.Branch)
+		if err != nil {
+			return objects.NewUploadObjectDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
+		}
+		if !branchExists {
+			return objects.NewUploadObjectNotFound().WithPayload(responseError("branch not found"))
```

Comment:
Maybe specify the branch name?  E.g. one way this can happen is if I ended up with a different branch than the one I intended.